src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.cpp

```diff
@@ -56,170 +61,337 @@ void metric_identity_jacobian(
+
+    metric_identity_det_jac_times_inv_jac_impl(
+        det_jac_times_inverse_jacobian, make_not_null(&buffer),
+        make_not_null(&buffer_component), mesh, inertial_coords, jacobian);
+  } else {
+    metric_identity_det_jac_times_inv_jac_impl(det_jac_times_inverse_jacobian,
+                                               mesh, inertial_coords, jacobian);
+  }
+}
+
+template <size_t Dim>
+void metric_identity_jacobian_quantities(
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
         det_jac_times_inverse_jacobian,
-    const Mesh<2>& mesh,
-    const tnsr::I<DataVector, 2, Frame::Inertial>& inertial_coords,
-    const Jacobian<DataVector, 2, Frame::Logical, Frame::Inertial>&
-        jacobian) noexcept;
-template void metric_identity_jacobian(
-    gsl::not_null<
-        InverseJacobian<DataVector, 3, Frame::Logical, Frame::Inertial>*>
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        inverse_jacobian,
+    const gsl::not_null<
+        Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        jacobian,
+    const gsl::not_null<Scalar<DataVector>*> det_jacobian,
+    const Mesh<Dim>& mesh,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords) noexcept {
+  static_assert(Dim == 1 or Dim == 2 or Dim == 3,
+                "Only implemented for 1, 2, and 3d.");
+  destructive_resize_components(det_jac_times_inverse_jacobian,
+                                mesh.number_of_grid_points());
+  destructive_resize_components(inverse_jacobian, mesh.number_of_grid_points());
+  destructive_resize_components(det_jacobian, mesh.number_of_grid_points());
+  ASSERT(
+      alg::all_of(*jacobian,
+                  [&mesh](const DataVector& jac_component) noexcept {
+                    return jac_component.size() == mesh.number_of_grid_points();
+                  }),
+      "The Jacobian components must all be the same size as the number of grid "
+      "points on the mesh.");
+
+  if constexpr (Dim == 3) {
+    // use inverse Jacobian as buffer in computation
+    metric_identity_det_jac_times_inv_jac_impl(
         det_jac_times_inverse_jacobian,
-    const Mesh<3>& mesh,
-    const tnsr::I<DataVector, 3, Frame::Inertial>& inertial_coords,
-    const Jacobian<DataVector, 3, Frame::Logical, Frame::Inertial>&
-        jacobian) noexcept;
+        make_not_null(&get<0, 0>(*inverse_jacobian)),
+        make_not_null(&get<0, 1>(*inverse_jacobian)), mesh, inertial_coords,
+        *jacobian);
+  } else {
+    metric_identity_det_jac_times_inv_jac_impl(
+        det_jac_times_inverse_jacobian, mesh, inertial_coords, *jacobian);
+  }
+
+  // Now compute the determinant of the Jacobian, the inverse Jacobian, and the
+  // Jacobian.
+  if constexpr (Dim == 1) {
+    get(*det_jacobian) = get<0, 0>(*jacobian);
+    get<0, 0>(*inverse_jacobian) = 1.0 / get<0, 0>(*jacobian);
```

Comment:
Here in 1D you're computing all quantities to be consistent with `jacobian`, not with `det_jac_times_inv_jac`. I guess in 1D it may not matter, but in the spirit of being precise, I suggest adding a comment here pointing out this optimization of sorts.